### PR TITLE
refactor/versioning

### PR DESCRIPTION
**Pull request by BrunnerLivio**
- [x] Update Readme
- [x] Update docker image
- [x] Update add-version.sh script for timestamp folder generating
- [x] Rename older versions 
- [x] Update NPM package
- [x] Add #10 version
- [x] Update test cases

Fix for issue #11

### CODE REVIEW

File: bin/add-version.sh [hunk 1]
```diff
@@ -67,17 +67,12 @@ NEW_VERSION_FOLDER="${VERSIONS_FOLDER}/${VERSION_TO_ADD}"
...
 mkdir -p $NEW_VERSION_FOLDER
 
-# Check if file is already in the destination
-if [ "$(stat -c "%d:%i" $PROTOBUF_FILE)" != "$(stat -c "%d:%i" $NEW_PROTOBUF_PATH)" ]; then
-  cp $PROTOBUF_FILE $NEW_PROTOBUF_PATH
-fi
+cp $PROTOBUF_FILE $NEW_PROTOBUF_PATH
 
-docker run -it --rm --name my-maven-project -v "$HOME/.m2":/root/.m2 -v `pwd`:/usr/src/mymaven -w /usr/src/mymaven maven:3.2-jdk-8 mvn clean package exec:java -Dexec.mainClass="com.pokebattler.gamemaster.GenerateJSON" -Dexec.args="${PROTOBUF_FILE} versions/${VERSION_TO_ADD}/GAME_MASTER.json"
+java -Dexec.mainClass="com.pokebattler.gamemaster.GenerateJSON" -Dexec.args="${PROTOBUF_FILE} versions/${VERSION_TO_ADD}/GAME_MASTER.json"
```
Comment (BrunnerLivio) [inline bin/add-version.sh]: Getting the error

```bash

Usage: java [-options] class [args...]
           (to execute a class)
   or  java [-options] -jar jarfile [args...]
           (to execute a jar file)
...

```

@celandro you know how I the command should be called correctly to generate the GAME_MASTER.json?

File: bin/add-version.sh [hunk 2]
```diff
@@ -67,17 +67,12 @@ NEW_VERSION_FOLDER="${VERSIONS_FOLDER}/${VERSION_TO_ADD}"
...
 mkdir -p $NEW_VERSION_FOLDER
 
-# Check if file is already in the destination
-if [ "$(stat -c "%d:%i" $PROTOBUF_FILE)" != "$(stat -c "%d:%i" $NEW_PROTOBUF_PATH)" ]; then
-  cp $PROTOBUF_FILE $NEW_PROTOBUF_PATH
-fi
+cp $PROTOBUF_FILE $NEW_PROTOBUF_PATH
 
-docker run -it --rm --name my-maven-project -v "$HOME/.m2":/root/.m2 -v `pwd`:/usr/src/mymaven -w /usr/src/mymaven maven:3.2-jdk-8 mvn clean package exec:java -Dexec.mainClass="com.pokebattler.gamemaster.GenerateJSON" -Dexec.args="${PROTOBUF_FILE} versions/${VERSION_TO_ADD}/GAME_MASTER.json"
+java -cp target/pokemongo-game-master-2.28.1.jar com.pokebattler.gamemaster.GenerateJSON "${PROTOBUF_FILE} versions/${VERSION_TO_ADD}/GAME_MASTER.json"
```
Comment (BrunnerLivio) [inline bin/add-version.sh]: I get the following error:

```
Error: Could not find or load main class com.pokebattler.gamemaster.GenerateJSON
```

Anyone knows a fix?

by calling

```bash

docker build -t "$USER/pokemongo-game-master" .
docker run \
  -v versions:/var/lib/pokemongo-game-master/versions \
  -it "$USER/pokemongo-game-master" \
  -f ./versions/0.57.1/GAME_MASTER.protobuf -v "0.69.9"

```

**Comment by celandro** [comment]
I can look tonight. But you probably aren't including the jar file

**Comment by BrunnerLivio** [comment]
@celandro I've fixed it, but it is not really efficient at the moment. I'm currently calling `mvn clean package` inside the `bin/add-version.sh` file. The build process should be inside the Dockerfile, but somehow I could not get it running because of the `exec: ` part. Not really a java expert, maybe you know the issue.

**Comment by BrunnerLivio** [comment]
I've changed the `add-version.sh` script so it will generate the current timestamp and set it as version number. 

For example, when calling the `docker run` script, it will create a new folder with the current timestamp e.g. `20180710083743` (= "+%Y%m%d%H%M%S"). You can change the timestamp by setting the `-t` parameter. If `--latest` is set, it will automatically update the latest-version.txt and copy the files to the latest folder.

**Comment by BrunnerLivio** [comment]
Can someone review this PR? @celandro @dandesousa

File: bin/add-version.sh [hunk 3]
```diff
@@ -49,36 +49,44 @@ do
...
+    esac
 done
 
 if [ "${PROTOBUF_FILE}" == "" ]; then
     print_error_and_exit "Missing protobuf file"
 fi
 
-if [ "${VERSION_TO_ADD}" == "" ]; then
-    print_error_and_exit "Missing version to add"
+if [ "${TIMESTAMP}" == "" ]; then
```
Comment (celandro) [inline bin/add-version.sh]: would be better if the timestamp was grepped out of the file

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+## Prerequisites
+
+1. Game Master file can be found on Android devices at `<internal storage OR sd card>/Android/data/com.ninaticlabs.pokemongo/files/remote_config_cache`. Copy the Game Master file in this directory.
+
+2. Install [Docker](https://docs.docker.com/install/).
+
+## Generate GAME_MASTER.json
+
+You can generate a new GAME_MASTER.json using the following command.
+It will create a new folder `versions/0.69.9`.
```
Comment (celandro) [inline CONTRIBUTING.md]: this is now datestamp based not niantic versions

File: bin/add-version.sh (hunk 3 above)
Comment (BrunnerLivio) [reply bin/add-version.sh]: Oh missunderstood this. I thought we should not use the timestamp of the GAME_MASTER file. But it makes more sense this way, will change that

**Comment by BrunnerLivio** [comment]
Can you review it again @celandro ?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+docker build -t "$USER/pokemongo-game-master" .
+docker run \
+  -v "$(pwd)/versions:/var/lib/pokemongo-game-master/versions" \
+  -v "$(pwd)/GAME_MASTER.protobuf:/var/lib/pokemongo-game-master/GAME_MASTER.protobuf" \
+  -v "$(pwd)/.git:/var/lib/pokemongo-game-master/.git" \
+  -v "$HOME/.gitconfig:/root/.gitconfig" \
+  -it "$USER/pokemongo-game-master" \
+  -f /var/lib/pokemongo-game-master/GAME_MASTER.protobuf --latest
+
```
Comment (celandro) [inline CONTRIBUTING.md]: I will add a windows version of this command when I have time. The syntax for current and home directory is a little different

Comment (BrunnerLivio) [reply CONTRIBUTING.md]: Would be nice if you can do that. Do not want to bother around with windows if I don't have to :)

**Pull request merged**